lib/awful/placement.lua

```diff
@@ -540,6 +555,33 @@ attach = function(d, position_f, args)
     if parent then
         parent:connect_signal("property::geometry" , tracker)
     end
+
+    -- Create a way to detach a placement function
+    d:add_signal("property::detach_callback")
+    function d.detach_callback()
+        d:disconnect_signal("property::width"       , tracker)
+        d:disconnect_signal("property::height"      , tracker)
+        d:disconnect_signal("property::border_width", tracker)
+        if parent then
+            parent:disconnect_signal("property::geometry" , tracker)
+
+            if parent == d.screen then
+                if args.honor_workarea then
+                    parent:disconnect_signal("property::workarea", tracker)
+                end
+
+                if args.honor_padding then
+                    parent:disconnect_signal("property::padding", tracker)
+                end
+            end
+        end
+
+        if args.update_workarea then
+            d:disconnect_signal("property::geometry" , tracker_struts)
+            d:disconnect_signal("property::visible"  , tracker_struts)
+            capi.client.disconnect_signal("property::struts", tracker_struts)
+        end
+    end
```

Comment:
Your screen-change test cover this for the screens. Given everything else use the same code, then I assume it works. There is a "leak" if something is not detached. But it actually make sense not to collect those wiboxes/clients/screens until then, so I don't plan to fix it, it's a feature.